File: frontend/src/types/prediction.ts
export interface PredictionRequest {
  carpet_area_sqft: number;
  floor_num: number;
  bathroom_num: number;
  balcony_num: number;
  parking_count: number;
  Transaction: string;
  Furnishing: string;
  facing: string;
  Ownership: string;
  parking_type: string;
  location_grouped: string;
  overlooks_main_road: number;
  overlooks_garden_park: number;
  overlooks_pool: number;
}

export interface PredictionResponse {
  predicted_price: number;
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel


class PredictionRequest(BaseModel):
    carpet_area_sqft: float
    floor_num: int
    bathroom_num: int
    balcony_num: int
    parking_count: int
    Transaction: str
    Furnishing: str
    facing: str
    Ownership: str
    parking_type: str
    location_grouped: str
    overlooks_main_road: int
    overlooks_garden_park: int
    overlooks_pool: int


class PredictionResponse(BaseModel):
    predicted_price: float